src/datachain/lib/dc.py

```diff
@@ -557,7 +557,9 @@ def jmespath_to_name(s: str):
                 nrows=nrows,
             )
         }
-        return chain.gen(**signal_dict)  # type: ignore[misc, arg-type]
+        # disable prefetch if nrows is set
+        settings = {"prefetch": 0} if nrows else {}
```

Comment:
disabled `prefetch` when `nrows` is set.

Reply:
should we just let users disable it though?

Reply:
The fact that we are doing pre-downloading/pre-fetching is not going to be obvious to the users. So, finding the right settings to disable/enable them is going to be difficult.

Similarly, the default settings with prefetch is going to be suboptimal when you only want to read a few rows. In fact, I'd have to adjust example tests to disable `prefetch` as well, as we use `laion` dataset to read just a couple of rows.

That said, I do not have any strong opinion, though — please let me know what you think.